File: 1635-number-of-good-pairs/1635-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int result = 0;
        
        for (int i : nums) {
            if (map.containsKey(i)) {
                int x = map.get(i);
                map.put(i, x + 1);
            } else {
                map.put(i, 1);
            }
        }
        
        for (int x : map.values()) {
            result += (x * (x - 1)) / 2;
        }
        
        return result;
    }
}

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        dic=defaultdict(int)
        result=0
        for i in nums:
            dic[i]+=1
        for num in dic:
            result+=(dic[num]*(dic[num]-1))//2
        return result
        